app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/LoginActivityUITest.java

```diff
@@ -0,0 +1,188 @@
+import android.support.test.espresso.intent.rule.IntentsTestRule;
+import android.support.test.rule.ActivityTestRule;
+import android.util.Log;
+
+import org.hamcrest.Matcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import static android.support.test.espresso.Espresso.closeSoftKeyboard;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * UI Test for Login Activity
+ */
+
+public class LoginActivityUITest {
+    private String usernameInputTestText = "username";
+    private String passwordInputTestText = "password";
+
+//    @Rule
+//    public ActivityTestRule<LoginActivity> mActivityRule = new ActivityTestRule<>(
+//            LoginActivity.class);
+
+   @Rule
+   public IntentsTestRule<LoginActivity> intentsTestRule = new IntentsTestRule<>(LoginActivity.class);
+
+    /**
+     * Test username with empty string
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void usernameEmptyTest() throws Exception {
+        onView(withId(R.id.txtUsername)).check(matches(withText("")));
+    }
+
+    /**
+     * Test username text box with input
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void usernameInputTextTest() throws Exception {
+        onView(withId(R.id.txtUsername)).perform(typeText(usernameInputTestText));
+        closeSoftKeyboard();
+        onView(withId(R.id.txtUsername)).check(matches(withText(usernameInputTestText)));
+    }
+
+    /**
+     * Test password with empty string
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void passwordEmptyTest() throws Exception {
+        onView(withId(R.id.txtPassword)).check(matches(withText("")));
+    }
+
+    /**
+     * Test password text box with input
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void passwordInputTextTest() throws Exception {
+        onView(withId(R.id.txtPassword)).perform(typeText(passwordInputTestText));
+        closeSoftKeyboard();
+        onView(withId(R.id.txtPassword)).check(matches(withText(passwordInputTestText)));
+    }
+
+    /**
+     * Test successful login when sign in button is clicked
+     *
+     * TODO: THIS TEST USES LOCKS TO WAIT FOR ASYNC OPERATIONS TO COMPLETE, IT SEEMS VERY HACKERY...BETTER APPROACHES???
```

Comment:
I haven't figured out too carefully how this works yet, but what are the downsides to doing it this way? Were you successfully able to make the test fail at first? It's just a test, so if this way works and there's no downsides then we shouldn't worry about it.

Reply:
The synchronization code was removed since it was not needed due to fixes in code after it was introduced.  The downside to it was that it was essentially negating asynchronous processing.  It was needed at the time to prevent the test from running to completion before getting a result from the authentication code, which meant that no assertions were being evaluated.